op-node/metrics/metrics.go

```diff
@@ -0,0 +1,150 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	Namespace = "op_node"
+
+	RPCServerSubsystem = "rpc_server"
+	RPCClientSubsystem = "rpc_client"
+
+	BatchMethod = "<batch>"
+)
+
+var (
```

Comment:
Should these metrics really be globals? We technically can have multiple node instances in the same process. And we do that in the go e2e testing for example. So these metrics would get data from entirely different nodes.
A metrics struct that we can pass to the rpc server and other node main components to avoid such collision would be nice.

Reply:
Done, updated to pass around a `metrics` object.